src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1128,6 +1128,27 @@ int PeerReplayer::pre_sync_check_and_open_handles(
   return 0;
 }
 
+int PeerReplayer::sync_perms(const std::string& path) {
+  int r = 0;
+  struct ceph_statx tstx;
+
+  // sync the mode of the remote dir_root with that of the local dir_root
+  r = ceph_statx(m_local_mount, path.c_str(), &tstx, CEPH_STATX_MODE,
+		 AT_STATX_DONT_SYNC | AT_SYMLINK_NOFOLLOW);
```

Comment:
the mode sync happens at line 1143, the code on 1136 just tries to fetch the stat for the local path, maybe the comment on line 1135 be updated accordingly? or shift the comment to line 1143?

Reply:
The comment is not at all necessary. However, I decided to keep it there for general guidance.

Reply:
then it can be re-positioned to the top maybe

Reply:
moved comment as the method description